app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddSidewalkSurface.kt

```diff
@@ -0,0 +1,147 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.ANYTHING_UNPAVED
+import de.westnordost.streetcomplete.data.meta.SIDEWALK_SURFACE_KEYS
+import de.westnordost.streetcomplete.data.meta.hasCheckDateForKey
+import de.westnordost.streetcomplete.data.meta.removeCheckDatesForKey
+import de.westnordost.streetcomplete.data.meta.updateCheckDateForKey
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.OUTDOORS
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.WHEELCHAIR
+
+class AddSidewalkSurface : OsmFilterQuestType<SidewalkSurfaceAnswer>() {
+
+    override val elementFilter = """
+        ways with
+            highway ~ trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential
+            and area != yes
+            and motorroad != yes
+            and (
+                sidewalk = both or sidewalk = left or sidewalk = right or
+                (sidewalk:left = yes and sidewalk:right ~ yes|no|separate) or
+                (sidewalk:right = yes and sidewalk:left ~ yes|no|separate)
+            )
+            and (
+                ${SIDEWALK_SURFACE_KEYS.joinToString(" and ") {"!$it"}}
+                or sidewalk:both:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:both:surface older today -4 years
+                or sidewalk:left:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:left:surface older today -4 years
+                or sidewalk:right:surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and sidewalk:right:surface older today -4 years
+                or ${SIDEWALK_SURFACE_KEYS.joinToString(" or ") {"$it older today -8 years"}}
+                or (
+                  sidewalk:both:surface ~ paved|unpaved|cobblestone
+                  and !sidewalk:both:surface:note
+                  and !note:sidewalk:both:surface
+                )
+                or (
+                  sidewalk:left:surface ~ paved|unpaved|cobblestone
+                  and !sidewalk:left:surface:note
+                  and !note:sidewalk:left:surface
+                )
+                or (
+                  sidewalk:right:surface ~ paved|unpaved|cobblestone
+                  and !sidewalk:right:surface:note
+                  and !note:sidewalk:right:surface
+                )
```

Comment:
That kind of filter was added to the surface quest because at the time the quest was created, there were still (and are still) a lot of ways tagged with these generic/ambiguous values. This doesn't hold true for `sidewalk:*:surface` because it is barely tagged anyway so far - compared to `surface`. So this additional complexity is not worth it, IMO.